diff --git a/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/lock.zap b/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/lock.zap
index 877bc2f..e99ac97 100644
--- a/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/lock.zap
+++ b/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/lock.zap
@@ -5076,6 +5076,15 @@
               "maxInterval": 65344,
               "reportableChange": 0
             }
+          ],
+          "events": [
+            {
+              "name": "LockOperation",
+              "code": 2,
+              "mfgCode": null,
+              "side": "server",
+              "included": 1
+            }
           ]
         }
       ]
diff --git a/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h b/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h
--- a/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h
+++ b/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h
@@ -618,7 +618,7 @@
 	}
 
 // clang-format off
-#define GENERATED_EVENT_COUNT 7
+#define GENERATED_EVENT_COUNT 8
 #define GENERATED_EVENTS { \
   /* Endpoint: 0, Cluster: Basic Information (server) */ \
   /* EventList (index=0) */ \
@@ -633,6 +633,9 @@
   /* Endpoint: 0, Cluster: Time Synchronization (server) */ \
   /* EventList (index=6) */ \
   0x00000004, /* MissingTrustedTimeSource */ \
+  /* Endpoint: 1, Cluster: Door Lock (server) */ \
+  /* EventList (index=7) */ \
+  0x00000002, /* LockOperation */ \
 }
 
 // clang-format on
@@ -1024,8 +1027,8 @@
       .functions = chipFuncArrayDoorLockServer, \
       .acceptedCommandList = ZAP_GENERATED_COMMANDS_INDEX( 65 ), \
       .generatedCommandList = ZAP_GENERATED_COMMANDS_INDEX( 77 ), \
-      .eventList = nullptr, \
-      .eventCount = 0, \
+      .eventList = ZAP_GENERATED_EVENTS_INDEX( 7 ), \
+      .eventCount = 1, \
     },\
   { \
       /* Endpoint: 1, Cluster: Approach Direction (server) */ \
