--- a/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h
+++ b/applications/matter-aliro-door-lock-app/src/matter/zap_uwb/zap-generated/endpoint_config.h
@@ -62,7 +62,7 @@
 	}
 
 // This is an array of EmberAfAttributeMetadata structures.
-#define GENERATED_ATTRIBUTE_COUNT 205
+#define GENERATED_ATTRIBUTE_COUNT 208
 #define GENERATED_ATTRIBUTES                                                                                                                            \
 	{                                                                                                                                               \
 		/* Endpoint: 0, Cluster: Descriptor (server) */                                                                                         \
@@ -607,6 +607,14 @@
 			  ZAP_ATTRIBUTE_MASK(READABLE) }, /* FeatureMap */                                                                              \
 			{ ZAP_SIMPLE_DEFAULT(8), 0x0000FFFD, 2, ZAP_TYPE(INT16U),                                                                       \
 			  ZAP_ATTRIBUTE_MASK(READABLE) }, /* ClusterRevision */                                                                         \
+			/* Endpoint: 1, Cluster: Approach Direction (server) */                                                                         \
+			{ ZAP_SIMPLE_DEFAULT(7), 0x00000000, 1, ZAP_TYPE(BITMAP8),                                                                      \
+			  ZAP_ATTRIBUTE_MASK(WRITABLE) | ZAP_ATTRIBUTE_MASK(NONVOLATILE) |                                                              \
+				  ZAP_ATTRIBUTE_MASK(READABLE) }, /* ApproachDirection */                                                               \
+			{ ZAP_SIMPLE_DEFAULT(0), 0x0000FFFC, 4, ZAP_TYPE(BITMAP32),                                                                     \
+			  ZAP_ATTRIBUTE_MASK(READABLE) }, /* FeatureMap */                                                                              \
+			{ ZAP_SIMPLE_DEFAULT(1), 0x0000FFFD, 2, ZAP_TYPE(INT16U),                                                                       \
+			  ZAP_ATTRIBUTE_MASK(READABLE) }, /* ClusterRevision */                                                                         \
 	}
 
 // clang-format off
@@ -756,7 +764,7 @@
 // clang-format on
 
 // This is an array of EmberAfCluster structures.
-#define GENERATED_CLUSTER_COUNT 20
+#define GENERATED_CLUSTER_COUNT 21
 // clang-format off
 #define GENERATED_CLUSTERS { \
   { \
@@ -1019,16 +1027,29 @@
       .eventList = nullptr, \
       .eventCount = 0, \
     },\
+  { \
+      /* Endpoint: 1, Cluster: Approach Direction (server) */ \
+      .clusterId = 0x1349FC03, \
+      .attributes = ZAP_ATTRIBUTE_INDEX(205), \
+      .attributeCount = 3, \
+      .clusterSize = 7, \
+      .mask = ZAP_CLUSTER_MASK(SERVER), \
+      .functions = NULL, \
+      .acceptedCommandList = nullptr, \
+      .generatedCommandList = nullptr, \
+      .eventList = nullptr, \
+      .eventCount = 0, \
+    },\
 }
 
 // clang-format on
 
-#define ZAP_FIXED_ENDPOINT_DATA_VERSION_COUNT 18
+#define ZAP_FIXED_ENDPOINT_DATA_VERSION_COUNT 19
 
 // This is an array of EmberAfEndpointType structures.
 #define GENERATED_ENDPOINT_TYPES                                                                                       \
 	{                                                                                                              \
-		{ ZAP_CLUSTER_INDEX(0), 17, 25 }, { ZAP_CLUSTER_INDEX(17), 3, 34 },                                    \
+		{ ZAP_CLUSTER_INDEX(0), 17, 25 }, { ZAP_CLUSTER_INDEX(17), 4, 41 },                                    \
 	}
 
 // Largest attribute size is needed for various buffers
@@ -1041,7 +1062,7 @@
 #define ATTRIBUTE_SINGLETONS_SIZE (0)
 
 // Total size of attribute storage
-#define ATTRIBUTE_MAX_SIZE (59)
+#define ATTRIBUTE_MAX_SIZE (66)
 
 // Number of fixed endpoints
 #define FIXED_ENDPOINT_COUNT (2)
