diff --git a/applications/matter-aliro-door-lock-app/src/aliro/access_manager/access_manager_impl.cpp b/applications/matter-aliro-door-lock-app/src/aliro/access_manager/access_manager_impl.cpp
index ec087aa..a5c3505 100644
--- a/applications/matter-aliro-door-lock-app/src/aliro/access_manager/access_manager_impl.cpp
+++ b/applications/matter-aliro-door-lock-app/src/aliro/access_manager/access_manager_impl.cpp
@@ -615,11 +615,11 @@ void AccessManagerImpl::_HandleRangingSessionData(SessionContext sessionContext,
 		const auto result =
 			id ? Aliro::Uwb::Disambiguation::Disambiguator::Instance().TryGetLastResult(*id) : std::nullopt;
 		const char *sideStr = result.has_value() ? (result->IsFront() ? "FRONT" : "BACK") : "----";
-		LOG_INF("session %p | %-16s | %s", sessionContext.GetRaw(),
+		LOG_DBG("session %p | %-16s | %s", sessionContext.GetRaw(),
 			openAllowed ? "OPEN ALLOWED" : "OPEN NOT ALLOWED", sideStr);
 	}
 #else
-	LOG_INF("session %p | %s", sessionContext.GetRaw(), openAllowed ? "OPEN ALLOWED" : "OPEN NOT ALLOWED");
+	LOG_DBG("session %p | %s", sessionContext.GetRaw(), openAllowed ? "OPEN ALLOWED" : "OPEN NOT ALLOWED");
 #endif // CONFIG_DOOR_LOCK_ALIRO_UWB_QM35_FRONT_BACK_DETECTION
 
 	SetOpenAllowed(sessionContext, openAllowed);
