diff --git a/applications/matter-aliro-door-lock-app/src/aliro/init.cpp b/applications/matter-aliro-door-lock-app/src/aliro/init.cpp
index ba88c0d..0e3fc49 100644
--- a/applications/matter-aliro-door-lock-app/src/aliro/init.cpp
+++ b/applications/matter-aliro-door-lock-app/src/aliro/init.cpp
@@ -8,7 +8,7 @@
 #include "aliro/features.h"
 #include "aliro/types.h"
 #include "aliro/utils.h"
-#include "nfc/nfc_transport_rfal.h"
+#include <woz_nfc/transport.h>
 
 #include <reader_storage/reader.h>
 
@@ -120,7 +120,7 @@ int AliroInit()
 	AliroError ec = AliroStack::Instance().Init();
 	VerifyOrReturnValue(ec == ALIRO_NO_ERROR, EXIT_FAILURE, LOG_ERR("Aliro stack initialization failed"));
 
-	ec = Aliro::NfcTransportRfal::Instance().Init();
+	ec = WozNfc::Init();
 	if (ec != ALIRO_NO_ERROR) {
 		LOG_ERR("NFC transport initialization failed");
 	}
@@ -163,7 +163,7 @@ int AliroInit()
 
 int AliroStart()
 {
-	AliroError ec = NfcTransportRfal::Instance().Start();
+	AliroError ec = WozNfc::Start();
 	if (ec != ALIRO_NO_ERROR) {
 		LOG_ERR("NFC transport start failed");
 		return EXIT_FAILURE;
@@ -193,7 +193,7 @@ int AliroStop()
 {
 	int rc = EXIT_SUCCESS;
 
-	AliroError ec = NfcTransportRfal::Instance().Stop();
+	AliroError ec = WozNfc::Stop();
 	if (ec != ALIRO_NO_ERROR) {
 		LOG_ERR("NFC transport stop failed");
 	}
diff --git a/applications/matter-aliro-door-lock-app/src/aliro/interface_impl/session.cpp b/applications/matter-aliro-door-lock-app/src/aliro/interface_impl/session.cpp
index ff08e59..0b89529 100644
--- a/applications/matter-aliro-door-lock-app/src/aliro/interface_impl/session.cpp
+++ b/applications/matter-aliro-door-lock-app/src/aliro/interface_impl/session.cpp
@@ -7,7 +7,7 @@
 #include "aliro/interface.h"
 
 #include "aliro/access_manager/access_manager.h"
-#include "aliro/platform/nfc/nfc_transport_rfal.h"
+#include <woz_nfc/transport.h>
 
 #ifdef CONFIG_NCS_ALIRO_BLE_UWB
 #include <aliro_service/aliro_service.h>
@@ -18,7 +18,7 @@ namespace Aliro::Interface::Session {
 AliroError Send(ConnectionHandle handle, Data data)
 {
 	if (handle.IsNfc()) {
-		return NfcTransportRfal::Instance().Send(data);
+		return WozNfc::Send(data);
 	}
 #ifdef CONFIG_NCS_ALIRO_BLE_UWB
 	else if (handle.IsBle()) {
@@ -32,7 +32,7 @@ AliroError Send(ConnectionHandle handle, Data data)
 void HandleTermination(ConnectionHandle handle)
 {
 	if (handle.IsNfc()) {
-		NfcTransportRfal::Instance().Terminate();
+		WozNfc::Terminate();
 	}
 #ifdef CONFIG_NCS_ALIRO_BLE_UWB
 	else if (handle.IsBle()) {
diff --git a/applications/matter-aliro-door-lock-app/src/aliro/platform/nfc/CMakeLists.txt b/applications/matter-aliro-door-lock-app/src/aliro/platform/nfc/CMakeLists.txt
index e7f635c..7bf9968 100644
--- a/applications/matter-aliro-door-lock-app/src/aliro/platform/nfc/CMakeLists.txt
+++ b/applications/matter-aliro-door-lock-app/src/aliro/platform/nfc/CMakeLists.txt
@@ -7,4 +7,4 @@
 zephyr_include_directories(.)
 
 # Core RFAL transport implementation
-zephyr_library_sources(nfc_transport_rfal.cpp)
+zephyr_library_sources_ifdef(CONFIG_NFC_DRIVER_STM nfc_transport_rfal.cpp)
